src/vs/workbench/services/userDataProfile/browser/userDataProfileManagement.ts

```diff
@@ -0,0 +1,157 @@
+
+import { VSBuffer } from 'vs/base/common/buffer';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { basename, joinPath } from 'vs/base/common/resources';
+import { URI } from 'vs/base/common/uri';
+import { localize } from 'vs/nls';
+import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
+import { ILocalExtension, Metadata } from 'vs/platform/extensionManagement/common/extensionManagement';
+import { IExtensionsProfileScannerService } from 'vs/platform/extensionManagement/common/extensionsProfileScannerService';
+import { ExtensionType } from 'vs/platform/extensions/common/extensions';
+import { IFileService } from 'vs/platform/files/common/files';
+import { registerSingleton } from 'vs/platform/instantiation/common/extensions';
+import { ILogService } from 'vs/platform/log/common/log';
+import { IProgressService, ProgressLocation } from 'vs/platform/progress/common/progress';
+import { IUserDataProfilesService } from 'vs/platform/userDataProfile/common/userDataProfile';
+import { IExtensionManagementServerService, IWorkbenchExtensionManagementService } from 'vs/workbench/services/extensionManagement/common/extensionManagement';
+import { IHostService } from 'vs/workbench/services/host/browser/host';
+import { CreationOptions, IUserDataProfileManagementService, IUserDataProfileTemplate, PROFILES_CATEGORY } from 'vs/workbench/services/userDataProfile/common/userDataProfile';
+
+const DefaultOptions: CreationOptions = {
+	settings: true,
+	keybindings: true,
+	tasks: true,
+	snippets: true,
+	extensions: true,
+	uiState: true
+};
+
+export class UserDataProfileManagementService extends Disposable implements IUserDataProfileManagementService {
+	readonly _serviceBrand: undefined;
+
+	constructor(
+		@IUserDataProfilesService private readonly userDataProfilesService: IUserDataProfilesService,
+		@IFileService private readonly fileService: IFileService,
+		@IExtensionManagementServerService private readonly extensionManagementServerService: IExtensionManagementServerService,
+		@IWorkbenchExtensionManagementService private readonly extensionManagementService: IWorkbenchExtensionManagementService,
+		@IExtensionsProfileScannerService private readonly extensionsProfileScannerService: IExtensionsProfileScannerService,
+		@IHostService private readonly hostService: IHostService,
+		@IDialogService private readonly dialogService: IDialogService,
+		@IProgressService private readonly progressService: IProgressService,
+		@ILogService logService: ILogService
+	) {
+		super();
+	}
+
+	async createAndEnterProfile(name: string, options: CreationOptions = DefaultOptions): Promise<void> {
+		const promises: Promise<any>[] = [];
+		const newProfile = this.userDataProfilesService.createProfile(name);
+		await this.fileService.createFolder(newProfile.location);
+		if (options?.uiState) {
+			promises.push(this.fileService.copy(this.userDataProfilesService.currentProfile.globalStorageHome, newProfile.globalStorageHome));
```

Comment:
I think this kind of state migration is dangerous, esp. on Windows where the file might be locked.

Reply:
Or is this only used in Web?

Reply:
This was done like this for prototype. This will be also done as we discussed in further PRs.